node/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -367,3 +370,23 @@ retrieveBlocks' i conv prevH endH = lift (recvLimited conv) >>= \case
         if curH == endH
         then pure $ one block
         else over _Wrapped (block <|) <$> retrieveBlocks' (i+1) conv curH endH
+
+----------------------------------------------------------------------------
+-- Networking
+----------------------------------------------------------------------------
+
+-- | Expects sending message to exactly one node. Receives result or
+-- fails if no result was obtained (no nodes available, timeout, etc).
+enqueueMsgSingle ::
+       (MonadThrow m)
+    => (t2 -> (t1 -> t -> NonEmpty x) -> m (Map NodeId (m b)))
+    -> t2
+    -> x
+    -> m b
+enqueueMsgSingle enqueue msg conv = do
+    results <- enqueue msg (\_ _ -> one conv) >>=
+               waitForConversations
+    case toList results of
+        [] ->      throwM $ DialogUnexpected "enqueueMsgSingle: nobody responded"
+        (_:_:_) -> throwM $ DialogUnexpected "enqueueMsgSingle: more than one node responded"
```

Comment:
Misleading error messages here. It's not that nobody responded, it's that the queue decided not to contact anybody.

Reply:
Fair.

Reply:
Fixed